packages/react/src/components/form/Fieldset/Fieldset.module.css

```diff
@@ -5,25 +5,35 @@
   min-width: 0;
 }
 
-.spacing > :not(:first-child, :empty) {
-  margin-top: var(--fds-spacing-2);
+.fieldset.withSpacing {
+  display: flex;
+  flex-direction: column;
+  gap: var(--fds-spacing-2);
 }
 
 .description {
   color: var(--fds-semantic-text-neutral-subtle);
   font-weight: 400;
 }
 
-.readonly > .legend {
+.legend {
+  display: contents;
+}
+
+.fieldset.readonly .legendContent {
   display: inline-flex;
 }
 
-.readonly > .legend > .padlock {
+.fieldset.readonly .padlock {
   height: 1.2em;
   width: 1.2em;
 }
 
-.disabled > .legend,
-.disabled > .description {
+.fieldset.disabled .legend,
+.fieldset.disabled .description {
   color: var(--fds-semantic-border-neutral-subtle);
 }
```

Comment:
I think we can just use the classes directly here, no need for `.fieldset.disabled .legend` - In this case we can just do `.disabled .legend`

Reply:
It does change the specificity, so that's a valid argument. I guess we want to keep it as low as possible. I'll change it.